### ISSUE TITLE
firmware_map doesn't work on C3 and S3

### ISSUE THREAD

**Issue by floitsch** [toitlang/toit#1555]
```
******************************************************************************
Decoding by `jag`, device has version <2.0.0-alpha.74>
******************************************************************************
EXCEPTION error. 
ERROR
  0: firmware_map_             <sdk>/system/firmware.toit:204:3
  1: map                       <sdk>/system/firmware.toit:30:13
  2: checksum                  /home/flo/code/artemis/src/service/run/device.toit:48:3
  3: main                      /home/flo/code/artemis/src/service/run/device.toit:24:25
******************************************************************************
```

**Comment by kasperl** [comment]
It looks like we get out-of-memory from the partition mapping code. Hmm.

This might be related to https://github.com/espressif/esp-idf/issues/9841.

**Comment by kasperl** [comment]
This may have been fixed by upgrading to ESP-IDF v5.0.x. I'll check.
